### PR TITLE
Add throw on failed initialization

### PR DESCRIPTION
**Pull request by LTA-Thinking**
## Description
Adds a throw for failed search parameter initialization.

## Related issues
Addresses [AB#119273](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/119273)

## Testing
Manual testing

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch

### CODE REVIEW

**Review by mikaelweave — DISMISSED** [review]
🦾

File: src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs [hunk 1]
```diff
@@ -96,14 +96,17 @@ public async Task EnsureInitializedAsync(CancellationToken cancellationToken)
             catch
             {
                 _initialized = false;
+                throw;
```
Comment (feordin) [inline src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs:99]: Where is this caught?

Comment (LTA-Thinking) [reply src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs:99]: Line 279

**Review by mikaelweave — APPROVED** [review]
🦾

**Pull request merged**